### PR TITLE
[DOCS] Add documents for Kyuubi client advanced features plan_only.md

### PR DESCRIPTION
**Pull request by CauchyHu**
### Code of Conduct
 - [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### _Why are the changes needed?_

Documents for Kyuubi client advanced features plan_only.md

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image
- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/client/advanced/features/plan_only.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Plan Only Execution Mode
+
+plan only execution mode is implemented through Spark extensions, so it is only supported Spark engine so far.
+
+It can print Spark's parser, analyze or optimize phases execution plan, and supports session-level and operation-level parameter settings, which are used in the following ways.
```
Comment (ulysses-you) [inline docs/client/advanced/features/plan_only.md:22]: the `session-level` and `operation-level` sound a little confused. In general, we call them: `application-level` and `session-level`

**Review by ulysses-you — COMMENTED** [review]
thank you @CauchyHu

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

File: docs/client/advanced/features/plan_only.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Plan Only Execution Mode
+
+Plan only execution mode is implemented through Spark extensions, so it is only supported Spark engine so far.
```
Comment (yaooqinn) [inline docs/client/advanced/features/plan_only.md:20]: it seems Flink engine is also supported

Comment (CauchyHu) [reply docs/client/advanced/features/plan_only.md:20]: I will update this document again

**Pull request closed without merging**